Component update - Default dropdown

alexandercyu opened the pull request:
Default dropdown updated to be closer to design.

- Additional checkbox added for the "All Catalogs" option
- Checking "All Catalogs" allows all checkboxes in the list to become checked

NOTE: Changes have been made that differ from the design to facilitate the functionality.
- Title of dropdown is not replaced a checkbox.

These changes were made because the dropdown depends on the "aria-expanded" attribute of the button/anchor tag. Replacing the text inside the button with the checkbox interferes with checkbox function and replacing the button altogether interferes with dropdown function.

alexandercyu committed:
Update dropdown.html

Add checkbox for "All Catalogs"

alexandercyu committed:
Update dropdown-default-scripts.js

Add script for "All Catalogs" checkbox that will check all checkboxes in the list

alexandercyu committed:
Update dropdown.html

Add "All Catalogs" checkbox to the default dropdown

alexandercyu committed:
Update dropdown-default-scripts.js

Fix function to run "All Catalogs" checkbox.
- Checks all checkboxes in list
- Gets correct count of checked checkboxes for title

mishaschwartz reviewed (changes requested):
This is what happens when I click the dropdown:

![image](https://github.com/user-attachments/assets/efc928a0-ac4d-463e-bcef-7cce4cd86e98)

Please take a look and fix.

alexandercyu committed:
Merge branch 'main' into component-update

alexandercyu committed:
Merge branch 'main' into component-update

alexandercyu committed:
Merge branch 'component-update' of https://github.com/DACCS-Climate/stac-search-ui into component-update

alexandercyu commented:
> This is what happens when I click the dropdown:
> 
> ![image](https://private-user-images.githubusercontent.com/4380924/414767915-efc928a0-ac4d-463e-bcef-7cce4cd86e98.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA1MTk5ODEsIm5iZiI6MTc0MDUxOTY4MSwicGF0aCI6Ii80MzgwOTI0LzQxNDc2NzkxNS1lZmM5MjhhMC1hYzRkLTQ2M2UtYmNlZi03Y2NlNGNkODZlOTgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIyNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMjVUMjE0MTIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzFmMzYyZGNkMzUzMDAxODRmZmVhZjY4MWQyZDEwMmJkNGVjMjc5MTRhNTQzOWVlMWVhMzM0OTZhNWYwMjViZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.i7877oMH--z3gctDHwkAbOwgJbxn1naQi0wrdw3p2yA)
> 
> Please take a look and fix.

Conflicts resolved.  The above screenshot shouldn't happen anymore.

mishaschwartz reviewed (changes requested):
There's a bug in the way the number of catalogs is counted:

![image](https://github.com/user-attachments/assets/2002922e-c387-4876-a7b7-058a57e3b0fd)

Also, if the placeholder text is "All Catalogs", does that mean that all the catalogs should be selected by default?

alexandercyu committed:
Update dropdown-list.js

Removed unneeded code

alexandercyu committed:
Update dropdown-default-scripts.js

- Changed checkCheckboxCount() function so it correctly counts the checkboxes when they are checked.
- added condition to checkCheckboxCount() function so it will uncheck the "All Catalogs" checkbox if not all the checkboxes are checked.

alexandercyu commented:
> There's a bug in the way the number of catalogs is counted:
> 
> ![image](https://private-user-images.githubusercontent.com/4380924/417163835-2002922e-c387-4876-a7b7-058a57e3b0fd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDA1ODkyMDIsIm5iZiI6MTc0MDU4ODkwMiwicGF0aCI6Ii80MzgwOTI0LzQxNzE2MzgzNS0yMDAyOTIyZS1jMzg3LTQ4NzYtYTdiNy0wNThhNTdlM2IwZmQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMjZUMTY1NTAyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWNkMDliN2NkY2U0NTY2OGU4MTE0YTgyOTg0NDM0OWY3MWUyZGQ4MmY4ZDk5ZjlhYmJmNDEyNzQwMTc4OGU4MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.zUDbX8no-PVbMxW-oJqNhS-PhyY6PFm8vBMHCRF0qSo)
> 


Fixed bug in how checkboxes are counted.

Added condition to uncheck "All Catalogs" checkbox if not all checkboxes are checked.

> Also, if the placeholder text is "All Catalogs", does that mean that all the catalogs should be selected by default?

I think Shruti meant it as "all catalogs available." The placeholder text just prevents a default catalog value.

mishaschwartz commented:
> I think Shruti meant it as "all catalogs available."

I see, I think it has the potential to be confusing but let's deal with that later
